exist-core/src/test/java/org/exist/xquery/ContextItemTests.java

```diff
@@ -1,90 +1,100 @@
+/*
+ * eXist-db Open Source Native XML Database
+ * Copyright (C) 2001 The eXist-db Authors
+ *
+ * info@exist-db.org
+ * http://www.exist-db.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
 package org.exist.xquery;
 
-import com.evolvedbinary.j8fu.Either;
-import org.exist.EXistException;
-import org.exist.security.PermissionDeniedException;
-import org.exist.storage.BrokerPool;
-import org.exist.storage.DBBroker;
-import org.exist.test.ExistEmbeddedServer;
-import org.exist.xquery.value.Sequence;
+import org.exist.test.ExistXmldbEmbeddedServer;
 import org.junit.ClassRule;
 import org.junit.Test;
+import org.xmldb.api.base.XMLDBException;
+import org.xmldb.api.modules.XQueryService;
 
-import static com.evolvedbinary.j8fu.Either.Left;
-import static com.evolvedbinary.j8fu.Either.Right;
+import static junit.framework.TestCase.fail;
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
 
+/**
+ * Ensure absolute XPath expression will raise a static error
+ *
+ * @author <a href="mailto:juri@existsolutions.com">Juri Leino</a>
+ */
 public class ContextItemTests {
 
     @ClassRule
-    public static final ExistEmbeddedServer server = new ExistEmbeddedServer(true, true);
+    public static final ExistXmldbEmbeddedServer existEmbeddedServer = new ExistXmldbEmbeddedServer(true, true, true);
 
     @Test
-    public void staticFunctionNoContextItem() throws EXistException, PermissionDeniedException {
-        final String query =
-                "declare function local:x() {" +
-                "  //x" +
-                "};" +
-                "local:x()";
-
-        final Either<XPathException, Sequence> result = executeQuery(query);
-        assertXQueryError(ErrorCodes.XPDY0002, result);
+    public void declaredFunctionAbsoluteSlash() {
+        final String query = "declare function local:x() { /x }; local:x()";
+        final String expected =
+                "err:XPDY0002 Leading '/' selects nothing, ContextItem is absent in function body [at line 1, column 30]";
+        assertCompilationError(query, expected);
```

Comment:
Checking such complex error messages by String comparison is much more fragile than just checking for the correct Error Code. As the thing that matters here is the error code, please change this back to match the error codes.

Reply:
Well, you are right it is a little more fragile to test for the complete error message than just the error code. 
This was a deliberate choice. That way the error code, the error message and the presence of the error location are tested.

Reply:
Okay but that mixes concerns, these tests are about the presence or not of a Content Item. The assertions here don't need to worry about the message or location info... we have other tests for location info now (in your other PR ;-))

Reply:
I disagree, the tests are there for what they test. And that is
- is a static error thrown?
- is the error code the expected one
- is the location info present
- is the error message correct

Reply:
Okay, then I can't agree to it as is :-(

Reply:
`org/exist/xquery/UnknownAtomicTypeTest.java` already use this type of tests and they were approved and merged.

Reply:
I see that, but actually it could have been done better there as well. I apologise for not spotting that in the review of that PR https://github.com/eXist-db/exist/pull/3546